scipy/stats/_resampling.py

```diff
@@ -969,22 +995,23 @@ def _calculate_null_pairings(data, statistic, n_permutations, batch,
     if n_permutations >= n_max:
         exact_test = True
         n_permutations = n_max
+        batch = batch or int(n_permutations)
         # cartesian product of the sets of all permutations of indices
         perm_generator = product(*(permutations(range(n_obs_sample))
                                    for i in range(n_samples)))
+        batched_perm_generator = _batch_generator(perm_generator, batch=batch)
```

Comment:
In this case - exhaustive test of all permutations - we're not changing any behavior. These changed lines have essentially been moved up from below.